docs/recitations/reci6-deployment.md

```diff
@@ -0,0 +1,96 @@
+
+There's quite a lot to learn about CI/CD. If you're interested in learning more, [click here!](https://about.gitlab.com/topics/ci-cd/)
+
+## Task 0: Prep
+
+For this recitation, we'll be working with a basic web app built on Next.js that responds to HTTP requests. Your job on Tuesday during the ~Extreme Startup~ game will be to parse these requests and figure out what the correct response is. Today we'll focus on setting up the app so that everything is in place for the game.
+
+The first thing you'll need to do is fork this repo: [https://github.com/CMU-313/basic-web-app](https://github.com/CMU-313/basic-web-app). Follow the instructions in the readme to run the development server locally. 
+
+Next, you'll need to set up the Fly.io account that you'll use to host your app. Complete steps 1-3 of the instructions [here](https://fly.io/docs/hands-on/). 
+
+Finally, complete the first deployment of your app by following the instructions [here](https://fly.io/docs/languages-and-frameworks/nextjs/). Open your app with `fly open`. 
+
+This has already been a lot of instructions, so good job for getting this far!!!
+
+**Note: Fly.io offers a free tier and you should not be required to enter any credit card info for the purposes of this recitation. If you get a message saying that a credit card is required, you can complete the recitation using Vercel instead. See the backup instructions below.**
+
+## Task 1: Continuous Deployment
+
+Next, we'll setup CD. Our goal is to set up a GitHub action that links to our Fly.io account and automatically redeploys a new version of our app whenever a new change is pushed into the main branch.
+
+The steps to do this are as follows:
+
+1. Get a Fly API token with `flyctl auth token`.
+2. Go to your fork on GitHub and click Settings.
+3. Go to Secrets and create a Repository secret called `FLY_API_TOKEN` with the value of the token from the previous step. This is how GitHub will link to your Fly.io account.
+4. Double check that `fly.toml` (the config file fly.io uses for deployment) is not listed in your `.gitignore`.
+5. Create a new file `.github/workflows/fly.yml`.This is the file that will outline instructions for the GitHub action. 
+6. Fill the file with the contents listed under step 8 [here](https://fly.io/docs/app-guides/continuous-deployment-with-github-actions/).
+7. Commit and push your changes!
+
+Now let's take a step back and understand what we've done. First, we generated an authorization token and used it to connect Fly.io to our GitHub repo. Then, we created a `fly.yml` file setting up a GitHub action that automatically runs the deployment commands whenever a push is made to the main branch. See if you can figure out which lines do what in the code above!
+
+With this complete, you'll be able to quickly iterate and any changes you make and push to your fork will be automatically deployed. Cool!
```

Comment:
Have them change the header on the website, and verify that deployment reflects the change on push?

Reply:
Resolved in latest commit